engine/lib/tags.php

```diff
@@ -13,9 +13,9 @@
  *
  * @param string $string Comma-separated tag string
  *
- * @return array|false An array of strings, or false on failure
+ * @return array|false|mixed An array of strings, false if no data provided, or the original data if input was not a string
  */
-function string_to_tag_array($string) {
+function string_to_tag_array($string = false) {
```

Comment:
$input is fine. Pretty much anything except $string ;) I think we leave the `@param` as string. This is what the function is meant for, we're just handling idiosyncrasies in how it's commonly used.